### ISSUE TITLE
File `%f` not found by script

### ISSUE THREAD

**Issue by rspencer01** [nextcloud/workflow_script#19]
### Steps to reproduce
1. Set up a script that reads the file specified such as the following
```python
#!/usr/bin/env python3
from PIL import Image
import sys, os

print("Opening image",sys.argv[1])
f = open(sys.argv[1]).read()
```
2. Add rule along the lines of `/path/to/script.py %f >> /tmp/test_output`
3. Trigger rule

### Expected behaviour
The script should execute without error and a line should be logged to `/tmp/test_output`

### Actual behaviour
The following error is thrown
```python
  File "/path/to/script", line 20, in <module>
    f = open(sys.argv[1]).read()
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/oc_tmp_0f9O8Q-.png'
```

Vexingly this works if the path to my script is replaced by `head -c 10` for example.

## Server configuration detail

**Operating system:** Linux 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64

**Webserver:** Apache/2.4.29 (Ubuntu) (apache2handler)

**Database:** mysql 5.7.27

**PHP version:** 

7.2.19-0ubuntu0.18.04.1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, sodium, session, standard, apache2handler, mysqlnd, PDO, xml, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, igbinary, imagick, intl, json, exif, mysqli, pdo_mysql, Phar, posix, readline, redis, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 16.0.3 - 16.0.3.0

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
    [files_rightclick] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [README.md] => Array
                        (
                            [expected] => 
                            [current] => cf73849388838de5037624e53303618579b164ec69fd55834ce4c5332ae33f778839e36298cbcc8ede90620927e076dbb8883479754e4c09b0491b15bf7653f3
                        )

                )

        )

)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.2.0
 - admin_audit: 1.6.0
 - bookmarks: 1.0.8
 - bruteforcesettings: 1.4.0
 - calendar: 1.7.0
 - cloud_federation_api: 0.2.0
 - comments: 1.6.0
 - contacts: 3.1.3
 - dav: 1.9.2
 - encryption: 2.4.0
 - federatedfilesharing: 1.6.0
 - federation: 1.6.0
 - files: 1.11.0
 - files_pdfviewer: 1.5.0
 - files_rightclick: 0.13.0
 - files_sharing: 1.8.0
 - files_versions: 1.9.0
 - files_videoplayer: 1.5.0
 - firstrunwizard: 2.5.0
 - gallery: 18.3.0
 - issuetemplate: 0.5.0
 - logreader: 2.1.0
 - lookup_server_connector: 1.4.0
 - metadata: 0.9.0
 - nextcloud_announcements: 1.5.0
 - notifications: 2.4.1
 - oauth2: 1.4.2
 - passman: 2.3.2
 - password_policy: 1.6.0
 - privacy: 1.0.0
 - provisioning_api: 1.6.0
 - recommendations: 0.4.0
 - serverinfo: 1.6.0
 - sharebymail: 1.6.0
 - support: 1.0.0
 - survey_client: 1.4.0
 - systemtags: 1.6.0
 - theming: 1.7.0
 - twofactor_backupcodes: 1.5.0
 - updatenotification: 1.6.0
 - user_usage_report: 1.1.2
 - viewer: 1.0.0
 - workflow_script: 1.1.0
 - workflowengine: 1.6.0
Disabled:
 - activity
 - files_external
 - files_texteditor
 - files_trashbin
 - previewgenerator
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        ***REMOVED SENSITIVE VALUE***
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "16.0.3.0",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "installed": true,
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpdebug": true,
    "loglevel": 0,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpsecure": "tls",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "theme": "",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "mail_sendmailmode": "smtp",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "htaccess.RewriteBase": "\/",
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "app_install_overwrite": [
        "calendar"
    ]
}
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

## Client configuration

**Browser:** Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0

**Operating system:** Ubuntu 18.04

**Comment by blizzz** [comment]
> Vexingly this works if the path to my script is replaced by head -c 10 for example.

Right, so the placeholder appears to be replaced fine. The location as in the error output seems correct. Did you double check there is nothing else needed to use the parameter in python?

**Comment by rspencer01** [comment]
I am quite positive.  I think it may have something to do with the `/tmp/systemd-private-3b3f0...2-apache2.service-dg...e/tmp` folder which has some files that match the format of the filename I am expecting.  Is this some kind of sandboxing issue?

**Comment by blizzz** [comment]
Aye, on systemd-based distros the /tmp of a service is jailed in the specific folder (here: `/tmp/systemd-private-3b3f0...2-apache2.service-dg...e/`). Trying another location like `dev/shm/ ` (memory!) should work.

**Comment by jwetzell** [comment]
@rspencer01 did you ever get this resolved I'm having the same issue trying to pass the file into a python script, getting the same `No such file or directory found error` if I do something like `cat %f > /tmp/test_output` and add a file that triggers the flow everything runs just fine.

**Comment by rspencer01** [comment]
I did not unfortunately, and will not be pursuing this further.

**Comment by jwetzell** [comment]
@rspencer01 I did get a workaround up by doing something like `cp %f /tmp/file.txt` and then pointing the python script to the `/tmp/file.txt`

**Comment by blizzz** [comment]
Should be fixed with #71, please reopen if not.
